File: forgerock-auth/src/main/java/org/forgerock/android/auth/PKCE.java
/*
 * Copyright (c) 2019 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * Domain object to store PKCE related data.
 */
@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
@Getter(value = AccessLevel.PACKAGE)
class PKCE {

    private final String codeChallenge;
    private final String codeChallengeMethod;
    private final String codeVerifier;

}

File: forgerock-auth/src/main/java/org/forgerock/android/auth/PKCE.kt
/*
 * Copyright (c) 2019 - 2025 Ping Identity Corporation. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */
package org.forgerock.android.auth

/**
 * Domain object to store PKCE related data.
 */
class PKCE(val codeChallenge: String,
                    val codeChallengeMethod: String,
                    val codeVerifier: String)
